File: 0011/main.py
class Solution:
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
# method(1) 枚举,会超时
#        temp = 0
#        max0 = 0
#        for i in range(len(height)):
#            j = i
#            while(j<len(height)):
#                temp = min(height[i],height[j])*(j-i)
#                if(temp>max0): max0 = temp
#                j += 1
#        return max0
# method(2) 双指针,两边夹
        left = 0
        right = len(height) - 1
        ans = 0
        while(right!=left):
            temp = min(height[left], height[right])*(right - left)
            ans = max(temp, ans)
            if height[left] > height[right]:
                right -= 1
            else:
                left += 1
        return ans

File: 0011/main.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        if(height.size()<2) return 0;
        int left = 0, right = height.size()-1;
        int count = 0;
        while(left < right) {
            count = max(count, (right-left)*min(height[left],height[right]));
            if(height[left] < height[right]) {
                ++left;
            } else {
                --right;
            }
        }
        return count;
    }
};